src/journal-remote/journal-remote-main.c

```diff
@@ -1160,9 +1160,9 @@ static int run(int argc, char **argv) {
         }
 
         notify_message = NULL;
-        (void) sd_notifyf(false,
```

Comment:
> We should maintain consistency and either do it everywhere or nowhere.
> Since the function is expected to fail and we always ignore the return value, let's just unvoidify all calls.

I prefer to explicitly voidify it everywhere. We usually voidify e.g., `pager_open()`. Why `sd_notify()` is special??

Reply:
`(void)` makes sense when it's the exception: by including it, we say "normally we would look at the return value of this function, but here we want to proceed no matter what the result is". When 100% of calls in the codebase to a given function are annotated with `(void)`, then the annotations are just noise. We *could* change the function to return `void` and have the same effect. But usually we don't want to change the return value of the function for various reasons (external api as in this case, a few uses of the return value for example in tests, or just function style), and simply ignoring the return value of the function is the cleanest and most appropriate choice.

In the past, we started adding `(void)` in various places to avoid a warning from coverity. The warning was along the lines of "The return value of this function is checked in 80% of calls, but not here. This looks like an error.". We also use `(void)` for functions which are declared with `__attribute__((__warn_unused_result__))`. Annotating the call with `(void)` silences the warning. This is now broken with gcc, but still works with clang. (And we actually make use of it now that #20109 has been merged.) In both cases the same general principle applies: **exceptions** are annotated to show intent.

Actually, I think calls to `pager_open()` should be de-voidified too, and the same for other functions where we generally don't care about the return value.

Reply:
I added a commit to drop `(void)` before `pager_open()` too.

Reply:
Hm, still I am not a fan of dropping `(void)`. You recently merged the commit to add `(void)` for `close_nointr()`... I cannot find any consistency with the commit and commits in this PR...

Reply:
Most of the `close_nointr` calls (14 out of 17) checked the return code, so static analyzers like coverity and friends raise warnings. But if 100% of calls to a function do not check the return code, then it's fine to skip the void cast, as it's not an exception anymore.
In other words, void casting is useful to indicate that usually a return code is checked, but that one is a special exception.

Reply:
@bluca Yeah, I know and understand that point. Sorry, voidification of `close_nointr()` is not a good point to comment here. Please forget about that.

But, when we see code with
```
(void) func();
```
then, we can easily understand that `func()` may fail, but we can safely ignore the failure or we have no good way to handle the failure (but in that case, we should warn about the failure...).
```
foo();
```
then, we see `foo()` does not fail, or does not return any error code.
These rules are quite important to understand code, especially people who is not familiar with our code base.

So, I still prefer to voidify _all_ function calls where we ignore error code.

Moreover, I prefer to introduce `int notify_or_warn()` which warns on failure.

Reply:
Also, I think explicit voidification at all places suggests people to implement good error handling for each function calls.

(But, anyway, my opinion seems to be minor at least now, please merge this as is, if all you agree with dropping voidifications.)

@poettering WDYT?

Reply:
uh, i am unsure. on one hand i get zbigniew's points, but also, if we ignore return codes everywhere then maybe the function should return void in the first place?

to me (void) cast indicates: "hey, i know this returns a useful error code, but here at this place I am not interested". 

Now the question if sd_notify() actually returns a "useful error code". I'd argue it does actually, and hence I am tempted to to leave (void) in. but then there's log_error() and friends which actually do return an error code, but i am not sure it's actually useful ever. Calls to log_error() we typically do not cast to void. So question is if sd_notifyf() is like log_error() or is not?

Or in other words, I am not sure at all, can't make up my mind what is better.

Hence I'd probably just not codify any recommendation here but allow either?


maybe the summary could be? it's not clear cut? and hence allow both?

Reply:
sd_notify() with STATUS= is kinda like log_error(), it really doesn't matter. sd_notify() with READY=1 otoh is not so irrelevant, because it is more than informational, it will actually affect daemon start-up behaviour.

maybe given that we should actually fix our sd_notify() invocations that are more than informational to check errors and somehow propagate them?